getl/blocks/fileregistry/s3_prefix_scan.py

```diff
@@ -0,0 +1,153 @@
+            self.last_date = self._get_last_prefix_date(dataframe)
+
+        self.delta_table = DeltaTable(self.file_registry_path, self.spark)
+
+    @staticmethod
+    def _get_files_to_lift(dataframe: DataFrame) -> List[str]:
+        """Get a list of s3 paths from the file registry that needs to be lifted."""
+        data = (
+            dataframe.where(F.col("date_lifted").isNull()).select("file_path").collect()
+        )
+
+        return [row.file_path for row in data]
+
+    def _update_file_registry(self, list_of_rows: List[FileRegistryRow]) -> DataFrame:
+        """Update the file registry and do not insert duplicates."""
+        updates_df = self._rows_to_dataframe(list_of_rows)
+
+        # Update the file registry
+        return self.delta_table.insert_all(
+            updates_df, "source.file_path = updates.file_path"
+        )
+
+    def _get_new_s3_files(self, s3_path: str, suffix: str) -> List[FileRegistryRow]:
+        """Get all new files in s3 from start to now as a dataframe."""
+        list_of_rows = []
+
+        for date in self._get_dates_from(self.last_date):
+            base_s3path = S3Path(s3_path) / date.strftime(self.partition_format)
+
+            keys = list(base_s3path.glob(suffix))
+            LOGGER.info("Search prefix %s for files. Found: %s", base_s3path, len(keys))
+
+            list_of_rows.extend(FileRegistryRow(str(key), date, None) for key in keys)
+
+        return list_of_rows
+
+    def _get_dates_from(self, start: datetime) -> List[datetime]:
+        """Get prefixes between the given dates."""
+        end = datetime.now().date()
+        step = timedelta(days=1)
+
+        while start <= end:
+            yield start
+            start += step
+
+    def _get_last_prefix_date(self, dataframe: DataFrame) -> str:
+        """Return the latest date_lifted column from the dataframe."""
+        date = (dataframe.agg(F.max("prefix_date")).collect())[0][0]
+
+        if not date:
+            date = self.default_start
+
+        return date
+
+    def _create_file_registry(self):
+        """When there is now existing file registry create one."""
+        dataframe = self.spark.createDataFrame([], self.schema)
+        dataframe.write.save(
+            path=self.file_registry_path, format="delta", mode="overwrite"
+        )
+        self._create_hive_table()
+
+    def _create_hive_table(self):
+        hive = HiveTable(self.spark, self.hive_database_name, self.hive_table_name)
+        hive.create(
+            self.file_registry_path, db_schema=self.db_schema,
+        )
+
+    def _rows_to_dataframe(self, rows: List[FileRegistryRow]) -> DataFrame:
+        """Create a dataframe from a list of paths with the file registry schema."""
+        # data = [(row.file_path, row.prefix_date, row.date_lifted) for row in rows]
+        return self.spark.createDataFrame(rows, self.schema)
+
+    def _create_file_registry_path(self, s3_path: str) -> str:
```

Comment:
This will need to be remove, the file registry path should be expressed explicitly in the __init__.